packages/pwa-kit-react-sdk/src/ssr/universal/hooks/index.js

```diff
@@ -70,3 +71,45 @@ export const useOrigin = ({fromXForwardedHeader = false}) => {
     }
     return APP_ORIGIN
 }
+
+/**
+ * Blocks the navigation to run a provided function whenever there is a new page being navigated to
+ * The function must return false to unblock the navigation
+ *
+ * @param {function} func
+ * @returns {boolean} State that is set to true during blocking, false otherwise.
+ */
+export const useBlockNavigation = (func) => {
+    const {block, push, location} = useHistory()
+    const lastLocation = useRef()
+    const [isBlocked, setIsBlocked] = useState(false)
+    const funcRef = useRef()
+    const abortControllerRef = useRef(new AbortController())
+    funcRef.current = func
+    useEffect(() => {
+        const unblock = block((location, action) => {
+            if (location?.pathname !== lastLocation.current?.pathname && funcRef.current) {
+                lastLocation.current = location
+                abortControllerRef.current.abort()
+                abortControllerRef.current = new AbortController()
+                ;(async () => {
+                    setIsBlocked(true)
+                    if (
+                        !(await funcRef.current(
+                            location,
+                            action,
+                            abortControllerRef.current.signal
+                        ))
+                    ) {
+                        setIsBlocked(false)
+                        unblock()
+                        push(location)
+                    }
+                })()
+                return false
+            }
+        })
+    }, [location])
+
+    return isBlocked
```

Comment:
What if we return an object like this instead?

Previously, if someone were to call `useBlockNavigation`, they're allowed to name their variable anything they want:

```js
const block = useBlockNavigation(...) // this is allowed
```
Did you want them to name their variable the same each time? If yes, then let's return an object like this. I think it makes for better DX.

```suggestion
    return {isBlocked}
```

Reply:
I think we would want to provide a name for this variable, so that it doesn't conflict with the name for the state in the extension store